src/NuGet.Core/NuGet.PackageManagement/PackageDownloader.cs

```diff
@@ -98,6 +99,18 @@ public static class PackageDownloader
                 groups.Enqueue(localGroup);
                 groups.Enqueue(otherGroup);
 
+                HashSet<string> nameSpacePackageSources = downloadContext.PackageNamespacesConfiguration?.Match(packageIdentity.Id);
+
+                if (nameSpacePackageSources != null)
+                {
+                    var packageSourcesAtPrefix = string.Join(", ", nameSpacePackageSources);
+                    logger.LogDebug(string.Format(CultureInfo.CurrentCulture, Strings.PackageNamespacePrefixMatchFound, packageIdentity.Id, packageSourcesAtPrefix));
```

Comment:
We've had issues in the past where a string resource was used in multiple places, it added a new `{x}` argument, but not all places were updated. I don't know what anyone else in the team thinks, but in NuGet.Packaging what I did was created a `StringFormatter` class which does the `string.Format`, and then other places call that method. This way, if the number of parameters in the string change, there's one place to change the `string.Format`, and then the compiler will help you find all other references.

Reply:
Ok. Addressed.